skymp5-server/cpp/lib_espm/impl/MappedBuffer.cpp

```diff
@@ -0,0 +1,99 @@
+#include "MappedBuffer.h"
+
+#include <fmt/format.h>
+
+#ifndef WIN32
+#  include <iostream>
+
+// Linux
+#  include <fcntl.h>
+#  include <sys/mman.h>
+#  include <unistd.h>
+#endif
+
+namespace espm::impl {
+
+MappedBuffer::MappedBuffer(const fs::path& path)
+{
+  size_ = fs::file_size(path);
+#ifdef WIN32
+  fileHandle_ = CreateFileW(path.c_str(), GENERIC_READ, FILE_SHARE_READ, NULL,
+                            OPEN_EXISTING, FILE_FLAG_RANDOM_ACCESS, 0);
+  if (!fileHandle_) {
+    // generic vs system?
+    throw std::system_error(
```

Comment:
Up to you. It seems reasonable to use `std::system_error` here.

Reply:
stackoverflow has no specific answer for that, so I guess yeah, we'll use `system_category` here and later on.

https://stackoverflow.com/questions/7542822/should-i-use-system-category-or-generic-category-for-errno-on-unix

Reply:
UPD: according to https://stackoverflow.com/a/35578374, `std::system_error` should be always safe to use.

More specifically, `system_error` should always be used on Windows, `generic_error` _can_ be used for **POSIX** codes on Linux.